myhpom/tasks.py

```diff
@@ -0,0 +1,119 @@
+import json
+import requests
+from django.core.mail import send_mail
+from django.conf import settings
+from django.template.loader import get_template
+from django.template import Context
+from celery import shared_task
+from celery.task import Task
+from celery.signals import task_failure
+from myhpom.models import AdvanceDirective, DocumentUrl, CloudFactoryRun, CloudFactoryUnit
+
+
+@shared_task
+class CloudFactorySubmitAdvanceDirectiveRun(Task):
+    def run(self, ad_id, line_id, document_host, callback_url):
+        """
+        ## Arguments
+        * ad_id = the database id for the AdvanceDirective
+        * line_id = the id of the CloudFactory production line to submit this request to.
+        * document_host = the host (incl. protocol/scheme) to use for the DocumentURL
+        * callback_url = the URL to use with the callback request
+
+        ## Task Process
+        - create an expiring DocumentUrl for the AdvanceDirective
+        - submit the DocumentUrl to the CloudFactory (test) endpoint
+        - store the status of the submitted document
+        - if the AdvanceDirective no longer exists, abort the task without an Exception.
+            - the AdvanceDirective might no longer exist by the time celery picks it up,
+            - or the document it points to might have been removed.
+        - if an error occurs in this process, send support email.
+        """
+        cf_run = CloudFactoryRun.objects.create(line_id=line_id, callback_url=callback_url or "")
```

Comment:
Provide an initial state that, in the worst case, gives us a sense of where we were in this task?

Reply:
like an initial status='Initialized' or some such?

Reply:
Exactly - the state of this model would then be explicitly known.